py/tests/integration/test_use_cases.py

```diff
@@ -1,39 +1,97 @@
 # coding: utf-8
 
+import tempfile
+import os
+import shutil
 from nose.tools import assert_equals, assert_not_equals, with_setup
 import common
 from waiting import wait
+from py.tests import chain_downloader
+
+settings = common.test_settings(__name__.split(".")[-1])
 
 def test_syncing():
     """
     Test node syncing:
     Alice should be able to connect to peers on startup and download the
     blockchain up to the current height.
     """
+    test_settings = settings["test_syncing"]
+    chain_data_file = test_settings["chain_data"]
     # start Bob's node
-    bob_node = "dev1"
+    bob_node = test_settings["nodes"]["bob"]
     common.start_node(bob_node)
     bob_api = common.external_api(bob_node)
 
-    # Insert a block in Bob's chain
-    common.post_fake_block(bob_api)
-    wait(lambda: bob_api.get_top().height == 1, timeout_seconds=3, sleep_seconds=0.5)
+    # Insert some blocks in Bob's chain
+    blocks_to_post = test_settings["blocks_to_post"]
+    post_blocks(bob_api, blocks_to_post, chain_data_file)
     bob_top = bob_api.get_top()
-    assert_equals(bob_top.height, 1)
-    # Now Bob has 2 blocks: with height 0 (genesis) and 1
+    assert_equals(bob_top.height, blocks_to_post)
+    # Now Bob has blocks_to_post blocks
 
     # start Alice's node and let it connect with Bob's
-    alice_node = "dev2"
+    alice_node = test_settings["nodes"]["alice"]
     common.start_node(alice_node)
     alice_api = common.external_api(alice_node)
-    wait(lambda: alice_api.get_top().height == 1, timeout_seconds=3, sleep_seconds=0.5)
+    wait(lambda: alice_api.get_top().height == blocks_to_post, timeout_seconds=3, sleep_seconds=0.5)
     alice_top = alice_api.get_top()
-    assert_equals(alice_top.height, 1)
-    # Alice has 2 blocks, too
+    assert_equals(alice_top.height, blocks_to_post)
+    # Alice has blocks_to_post blocks, too
 
-    # ensure that both nodes have the same hash on block with height=1
+    # ensure that both nodes have the same hash on top 
     assert_equals(alice_top.hash, bob_top.hash)
 
     # stop both nodes
     common.stop_node(bob_node)
     common.stop_node(alice_node)
+
+def test_persistence():
+    """
+    Test persistence:
+    Bob's downloaded blockchain should persist between restarts. He should
+    only download updates to his blockchain when his node starts.
+    """
+    test_settings = settings["test_persistence"]
+    chain_data_file = test_settings["chain_data"]
+
+    # prepare a dir to hold the config and DB files 
+    root_dir = tempfile.mkdtemp()
```

Comment:
the erlang node expects those directories? (it is not creating them?)

Reply:
We need a place to put the `.chain` - containing the DB files; I didn't want to interfere with any data currently in the build, so I've created a temporary one

Reply:
ok, yes, nev